### ISSUE TITLE
SQLException: Access denied for user 'test'@'172.17.0.1' (using password: YES)

### ISSUE THREAD

**Issue by itay-keller** [testcontainers/testcontainers-java#718]
I'm trying to create an integration test for my Spring Boot app with MySQL container.

So this is the CatIntegrationTests class:

    `
    @RunWith(SpringRunner.class)
    @SpringBootTest(classes = CatApplication.class,
        webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
    @TestPropertySource(locations = "classpath:application-integrationtests.properties")
    @AutoConfigureMockMvc
     public class CatIntegrationTest {   
      @LocalServerPort
       private int port;
       TestRestTemplate restTemplate = new TestRestTemplate();

       @Autowired
       private MockMvc mvc;

    MySQLContainer mySQLContainer;

    @Before
    public void before(){
        mySQLContainer = new MySQLContainer("mysql:5.7.22").withDatabaseName("cats").withUsername("test").withPassword("test");
        mySQLContainer.start();
    }

    @After
    public void after(){
        mySQLContainer.stop();

    }
`

And the `application-integrationtests.properties` is like that:

> spring.jpa.hibernate.ddl-auto=create
> spring.jpa.database=mysql
> spring.datasource.url=jdbc:mysql://localhost:3306/cats
> spring.datasource.username=test
> spring.datasource.password=test
> spring.datasource.driver-class-name=com.mysql.jdbc.Driver

When trying to run the integration test, I get this error (and I really don't see the container spinning up):

`java.sql.SQLException: Access denied for user 'test'@'172.17.0.1' (using password: YES)`

Any idea what is missing here?

**Comment by bsideup** [comment]
Hi @itay-keller,

Since I had it in my workshop, I hope it's fine if I refer it here? :)
https://bsideup.gitbooks.io/spring-io-testcontainers-workshop/content/steps/step4.html

TL;DR:
1) use JDBC url (see the link, you will find an explanation and also a reference to the docs)
2) configure your properties properly, MySQL **will not** be running on localhost:3306

**Comment by itay-keller** [comment]
Hi @bsideup .
It worked!!! Big big thanks!!

**Comment by bsideup** [comment]
@itay-keller woooo 🎉  I'm closing the issue as "invalid", but feel free to report more 👍
